R/tbl_reg_summary.R

```diff
@@ -0,0 +1,69 @@
+#' Data Summary Table
+#'
+#' Function wraps `gtsummary::tbl_summary()` to create a data summary
+#' table often seen in regulatory submissions. Continuous variable summaries
+#' are shown on multiple lines with additional summary statistics and percentages
+#' are shown with more precision for categorical variables.
```

Comment:
```suggestion
#' table often seen in regulatory submissions. Continuous variable summaries
#' are shown on multiple lines with additional summary statistics and percentages
#' are shown for categorical variables; precision levels estimated based on values observed.
```
"More" precision is a bit confusing to me here.  More than what?

Reply:
oh good point, the reference is `tbl_summary()`, but that is never stated!

I think this update makes it sound like the level of precision for categorical variables is selected based on the observed values, but it's static: only the continuous variables have dynamic default rounding.

Reply:
Sorry if I am confused! With `tbl_summary-fn:percent_fun" = function(x) style_percent(x, digits = 1)`, doesn't this give dynamic decimals for categorical variables? Like it is either 1 or 2 decimals?

Reply:
oh yes, 3 significant figs, e.g. `90.4`, 3.45`

Reply:
Regarding theme documentation, how about something like this?

This method is executed via a gtreg theme; other custom or pre-specified themes executed will not overwrite the `tbl_reg_summary()` formatting.

Reply:
I'm not sure the best way to communicate this. On the one hand, I like the transparency of being vv clear with users what is going on. But I also think mentioning themes will confuse more users than it will help?

Also, since we're implementing this function by simply calling a theme, a user who uses themes should just use their theme on tbl_summary()?